src/reducers/notifications.ts

```diff
@@ -176,10 +188,26 @@ const messages: {[key: string]: (state: IRootState, options: any) => string} = {
  */
 export const notificationUpdater: Middleware =
   ({ getState, dispatch }) =>
-  (next) => (action: any) => {
+  (next) => {
+    const transaction2Notification = (network: string, id: string, {error, status, functionName, options, txHash, proposalTitle}: IOperation) => {
+      const actionMessage = messages[functionName] && messages[functionName](proposalTitle, options);
+      const errorReason = error ?
+        (
+          error === OperationError.Canceled ?
+            'you canceled the tx' :
+          error === OperationError.Reverted ?
+            'the tx was reverted' :
+          error === OperationError.OutOfGas ?
+            'of insufficient gas' :
+            `of unknown reason, contact us at https://discord.gg/3d5C2y7`
+        ) : '';
+
+      let fullErrorMessage;
+      if (error && error !== OperationError.Canceled && error !== OperationError.Canceled && error !== OperationError.Canceled) {
```

Comment:
uh, you used the same condition three times

Reply:
Thanks!
https://github.com/daostack/alchemy/pull/119/commits/771462e487ce5a8046e35910f854c57c9fd61858